service/src/test/java/org/cbioportal/service/impl/MutationCountServiceImplTest.java

```diff
@@ -0,0 +1,187 @@
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.cbioportal.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.cbioportal.model.PositionMutationCount;
+import org.cbioportal.persistence.MutationCountRepository;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ *
+ * @author abeshoua
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class MutationCountServiceImplTest {
+	@InjectMocks
+    private MutationCountServiceImpl mutationCountService;
+
+    @Mock
+    private MutationCountRepository mutationCountRepository;
+    
+    @Test
+    public void getPositionMutationCounts() throws Exception {
+
+        ArrayList<String> testGenes = new ArrayList<>();
+	testGenes.add("BRAF");
+	testGenes.add("KRAS");
+        ArrayList<List<Integer>> testPositions = new ArrayList<>();
+	ArrayList<Integer> brafPositions = new ArrayList<>();
+	ArrayList<Integer> krasPositions = new ArrayList<>();
+	brafPositions.add(1);
+	brafPositions.add(2);
+	krasPositions.add(3);
+	testPositions.add(brafPositions);
+	testPositions.add(krasPositions);
+
+	PositionMutationCount pmc1 = new PositionMutationCount();
+	PositionMutationCount pmc2 = new PositionMutationCount();
+        List<PositionMutationCount> brafResult = new ArrayList<>();
+	List<PositionMutationCount> krasResult = new ArrayList<>();
+	brafResult.add(pmc1);
+	krasResult.add(pmc2);
+
+        Mockito.when(mutationCountRepository.getPositionMutationCounts(org.mockito.Matchers.eq("BRAF"), org.mockito.Matchers.anyList())).thenReturn(brafResult);
+	Mockito.when(mutationCountRepository.getPositionMutationCounts(org.mockito.Matchers.eq("KRAS"), org.mockito.Matchers.anyList())).thenReturn(krasResult);
+	
+	List<PositionMutationCount> expectedResult = new ArrayList<>();
+	expectedResult.add(pmc1);
+	expectedResult.add(pmc2);
+	List<PositionMutationCount> result = mutationCountService.getPositionMutationCounts(testGenes, testPositions);
+	Assert.assertEquals(result, expectedResult);
+    }
+    
+    @Test
+    public void getPositionMutationCountsOnePositionListMissing() throws Exception {
+
+        ArrayList<String> testGenes = new ArrayList<>();
+	testGenes.add("BRAF");
+	testGenes.add("KRAS");
+        ArrayList<List<Integer>> testPositions = new ArrayList<>();
+	ArrayList<Integer> brafPositions = new ArrayList<>();
+	brafPositions.add(1);
+	brafPositions.add(2);
+	testPositions.add(brafPositions);
```

Comment:
so we are calling without supplying the kras position list? Then we expect only the braf result coming out.
I don't understand why this functionality should be supported. I think there is a concept I am missing here